be/src/exec/spill/spiller.hpp

```diff
@@ -165,21 +171,22 @@ Status SpillerReader::trigger_restore(RuntimeState* state, TaskExecutor&& execut
     DCHECK(_stream->enable_prefetch());
     // if all is well and input stream enable prefetch and not eof
     if (!_stream->eof()) {
+        _running_restore_tasks++;
         auto restore_task = [this, state, guard]() {
             RETURN_IF(!guard.scoped_begin(), Status::OK());
-            _running_restore_tasks++;
-            SerdeContext ctx;
-            auto res = _stream->prefetch(ctx);
-
-            if (!res.is_end_of_file() && !res.ok()) {
-                _spiller->update_spilled_task_status(std::move(res));
-            }
-
+            auto defer = DeferOp([&]() { _running_restore_tasks--; });
```

Comment:
the `defer` should before `RETUNR_IF`

Reply:
it shouldn't be before RETURN_IF. SpillReader maybe released